activerecord/test/cases/timestamp_test.rb

```diff
@@ -457,6 +464,14 @@ def test_index_is_created_for_both_timestamps
   ensure
     ActiveRecord::Base.connection.drop_table(:foos)
   end
+
+  def test_datetime_precision_applied_to_timestamp
+    d = Parrot.create!(name: 'Bogdan')
+    in_instance = d.created_at
+    from_db = Parrot.find(d.id).created_at
+    assert_equal from_db.usec, in_instance.usec
+    assert_equal from_db, in_instance
```

Comment:
There are already several tests of this nature in `activerecord/test/cases/type`